src/gas/GasFeeController.test.ts

```diff
@@ -99,25 +102,76 @@ describe('GasFeeController', () => {
 
   afterEach(() => {
     nock.cleanAll();
+    jest.clearAllMocks();
     gasFeeController.destroy();
   });
 
   it('should initialize', async () => {
     expect(gasFeeController.name).toBe(name);
   });
 
-  it('should getGasFeeEstimatesAndStartPolling', async () => {
-    expect(gasFeeController.state.gasFeeEstimates).toStrictEqual({});
-    const result = await gasFeeController.getGasFeeEstimatesAndStartPolling(
-      undefined,
-    );
-    expect(result).toHaveLength(36);
-    expect(gasFeeController.state.gasFeeEstimates).toHaveProperty('low');
-    expect(gasFeeController.state.gasFeeEstimates).toHaveProperty('medium');
-    expect(gasFeeController.state.gasFeeEstimates).toHaveProperty('high');
-    expect(gasFeeController.state.gasFeeEstimates).toHaveProperty(
-      'estimatedBaseFee',
-    );
+  describe('getGasFeeEstimatesAndStartPolling', () => {
+    it('should fetch estimates and start polling', async () => {
+      expect(gasFeeController.state.gasFeeEstimates).toStrictEqual({});
+      const result = await gasFeeController.getGasFeeEstimatesAndStartPolling(
+        undefined,
+      );
+      expect(result).toHaveLength(36);
+      expect(gasFeeController.state.gasFeeEstimates).toHaveProperty('low');
+      expect(gasFeeController.state.gasFeeEstimates).toHaveProperty('medium');
+      expect(gasFeeController.state.gasFeeEstimates).toHaveProperty('high');
+      expect(gasFeeController.state.gasFeeEstimates).toHaveProperty(
+        'estimatedBaseFee',
+      );
+    });
+
+    it('should not fetch estimates if the controller is already polling, and should still return the passed token', async () => {
+      const pollToken = 'token';
+
+      const firstCallPromise = gasFeeController.getGasFeeEstimatesAndStartPolling(
+        undefined,
+      );
+      const secondCallPromise = gasFeeController.getGasFeeEstimatesAndStartPolling(
+        pollToken,
+      );
+
+      const result1 = await firstCallPromise;
+      const result2 = await secondCallPromise;
+
+      expect(mockRequestHandler).toHaveBeenCalledTimes(1);
+      expect(result1).toHaveLength(36);
+      expect(result2).toStrictEqual(pollToken);
+    });
+
+    it('should fetch new estimates if the poll tokens are cleared, and then should not make additional new fetches', async () => {
```

Comment:
I'm a bit confused on this one... I see it that it shouldn't make additional fetches on getGasFeeEstimatesAndStartPolling calls (until pollTokens are cleared again), but won't it make additional fetches on the polling interval?

Reply:
I guess this comment can be clarified. It means to say that subsequent calls to getGasFeeEstimatesAndStartPolling should not trigger new fetches

Reply:
@adonesky1 updated the comment in the latest commit